compiler/noirc_frontend/src/monomorphization/mod.rs

```diff
@@ -1590,3 +1783,30 @@ fn undo_instantiation_bindings(bindings: TypeBindings) {
         var.unbind(id);
     }
 }
+
+fn get_field(ptype: &PrintableType, field_name: &str) -> Option<usize> {
+    match ptype {
+        PrintableType::Struct { fields, .. } => {
+            fields.iter().position(|(name, _)| name == field_name)
+        }
+        PrintableType::Tuple { .. } | PrintableType::Array { .. } => {
+            field_name.parse::<usize>().ok()
+        }
+        _ => None,
+    }
+}
+
+fn next_type(ptype: &PrintableType, i: usize) -> PrintableType {
+    match ptype {
+        PrintableType::Array { length: _length, typ } => (**typ).clone(),
+        PrintableType::Tuple { types } => types[i].clone(),
+        PrintableType::Struct { name: _name, fields } => fields[i].1.clone(),
+        PrintableType::String { length: _length } => {
+            // TODO: check bounds
+            PrintableType::UnsignedInteger { width: 8 }
+        }
+        _ => {
+            panic!["expected type with sub-fields, found terminal type"]
+        }
+    }
+}
```

Comment:
I think `element_type_at_index` would be a better name here. It also looks like this `TODO` wouldn't be terribly difficult to fill in. I also think it's fine to remove though since the only element type of a String would be a character anyway.

Reply:
I think this refers to checking that the `i` index accessed is within the length of the string. This is needed in the context of assigning to elements of a string, which I don't think is possible in Noir, correct?

I'm not sure it's a known issue, but I get a weird error when trying to do:
```noir
fn main() {
    let mut s = "hello";
    s[0] = s[1];
}
```
Here's the error:
```
error: Expected type array, found type str<5>
  ┌─ /home/ggiraldez/Scratchpad/mutable_str/src/main.nr:3:13
  │
3 │     s[10] = 64;
  │             --
  │
```

Reply:
Hmm, the location of that error is wrong. It's trying to say it expected `s` to be an array to be accessed by index like that. This could be a good issue to file. Currently strings don't support many operations at all and are mostly for debug printing purposes.

Reply:
> I think this refers to checking that the i index accessed is within the length of the string. This is needed in the context of assigning to elements of a string, which I don't think is possible in Noir, correct?

I don't think it should be required for `next_type` specifically though since the only type for strings is the character type. Similar to the array case where we don't check the size either.

Reply:
Agreed and reflected in the last commits. I'll file an issue later on the error location.